web/client-api/src/main/java/io/deephaven/web/client/api/barrage/BarrageUtils.java

```diff
@@ -116,8 +116,7 @@ public static TableSnapshot createSnapshot(BarrageRecordBatch header, ByteBuffer
         Iter<Buffer> buffers = new Iter<>(IntStream.range(0, (int) header.buffersLength()).mapToObj(header::buffers).iterator());
         ColumnData[] columnData = new ColumnData[0];
         for (int columnIndex = 0; columnIndex < columnTypes.length; ++columnIndex) {
-            columnData[columnIndex] = readArrowBuffer(body.slice(), nodes, buffers, (int) includedAdditions.size(), columnTypes[columnIndex]);
-            assertAligned(body);
+            columnData[columnIndex] = readArrowBuffer(body, nodes, buffers, (int) includedAdditions.size(), columnTypes[columnIndex]);
```

Comment:
Q: The goal here was that readArrowBuffer might modify body's position/offset. Can we guarantee that no unwanted side effects are working here?

Reply:
Since all offsets are absolute in the body, this is totally safe - every caller who wants a specific position should specify it.